js/app/historyStore.js

```diff
@@ -119,6 +122,29 @@ const HistoryStore = new Lang.Class({
 
     // Default signal handler
     on_changed: function () {
+        let old_item = this._direction == Direction.FORWARDS ?
+            this.get_previous_item() : this.get_next_item();
+        let item = this.get_current_item();
+
+        if (!(item.model instanceof Eknc.MediaObjectModel) &&
+            old_item && old_item.model && old_item.page_type == Pages.ARTICLE &&
+            (!item.model || old_item.model.ekn_id !== item.model.ekn_id))
+            Utils.record_content_access_metric(false,
+                                               '',
+                                               old_item.model.ekn_id,
+                                               old_item.model.title,
+                                               old_item.model.content_type);
+
+        if (item.model && item.model instanceof Eknc.ArticleObjectModel &&
+            (!old_item || !old_item.model ||
+             old_item.model.ekn_id !== item.model.ekn_id)) {
+            Utils.record_content_access_metric(true,
+                                               item.entry_point ? item.entry_point : '',
+                                               item.model.ekn_id,
+                                               item.model.title,
```

Comment:
In theory they are guaranteed, but content is messy, so I would prefer to be defensive here.